### PR TITLE
Trigger host restart when encrypting

### PR DESCRIPTION
**Pull request by mpolden**
@hakonhall

### CODE REVIEW

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/node/Nodes.java [hunk 1]
```diff
@@ -641,7 +641,9 @@ private void requireBreakfixable(Node node) {
             Node newHost = host.withWantToRetire(true, wantToDeprovision, wantToRebuild, agent, instant);
             if (op == DecommissionOperation.encrypt) {
                 Report report = Report.basicReport(Report.WANT_TO_ENCRYPT_ID, Report.Type.UNSPECIFIED, instant, "");
-                newHost = newHost.with(newHost.reports().withReport(report));
+                newHost = newHost.with(newHost.reports().withReport(report))
+                                 // Restart host-admin to trigger run of single-resume disk tasks
+                                 .withRestart(newHost.allocation().get().restartGeneration().withIncreasedWanted());
```
Comment (hakonhall) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/node/Nodes.java:646]: The restart must be triggered only after it reaches "parked".

Comment (mpolden) [reply node-repository/src/main/java/com/yahoo/vespa/hosted/provision/node/Nodes.java:646]: Of course, thanks!

**Comment by mpolden** [comment]
PTAL

File: node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/HostEncrypter.java [hunk 2]
```diff
@@ -95,6 +97,12 @@ private int hostLimit(NodeList hosts, NodeType hostType) {
         return Math.max(0, limit - hosts.encrypting().size());
     }
 
+    /** Trigger restart of encrypting nodes to allow disk encryption to happen */
+    private void triggerRestart(NodeList allNodes, NodeType nodeType) {
+        NodeList hostsReadyToEncrypt = allNodes.nodeType(nodeType).state(Node.State.parked).encrypting();
+        nodeRepository().nodes().restart(NodeListFilter.from(hostsReadyToEncrypt.asList()));
```
Comment (hakonhall) [inline node-repository/src/main/java/com/yahoo/vespa/hosted/provision/maintenance/HostEncrypter.java:103]: Only restart if not already restarting?

**Review by hakonhall — APPROVED** [review]

**Review by hakonhall — APPROVED** [review]

**Pull request merged**